### DISCUSSION TITLE
filament relation data in table

### DISCUSSION THREAD

**Discussion by Youssef-Elouasbi** [filamentphp/filament#12476] [Help]
### Package

Panel builder

### Package Version

3.2.61

### How can we help you?

i have this model with this relation and now i can see fields from table enseignes in my relation table but i want to see fields from table "convention_enseignes" how can i do that ? please i need help

class Convention extends Model
{

    protected $table = 'conventions';

    protected $fillable = [
        'nom', 'account_id', 'pourcentage_remise', 'date_debut', 'date_fin', 'statut_id', 'type_id', 'cree_par', 'modifie_par', 'proprietaire'
    ];

   public function enseignes(): BelongsToMany
    {
        return $this->belongsToMany(Enseigne::class, 'convention_enseignes')->withPivot('cree_par', 'modifie_par', 'proprietaire', 'date_debut', 'date_fin')->withTimestamps();
    }
}

---------------------------------
class ConventionEnseigne extends Model
{
 
    protected $table = 'convention_enseignes';

    protected $fillable = [
        'convention_id', 'enseigne_id', 'description', 'date_debut', 'date_fin', 'cree_par', 'modifie_par', 'proprietaire'
    ];
}
---------------------------------------

class Enseigne extends Model
{
    protected $table = 'enseignes';
   protected $fillable = [
        'account_id', 'nom', 'date_debut', 'solde', 'description', 'secteur_activite_id',
        'date_fin', 'statut_id', 'picture', 'point_price', 'logo', 'percentage_remise', 'url',
        'encours_max',
        'cree_par', 'modifie_par', 'proprietaire', 'categorie_id'
    ];
}

**Comment by leandrocfe** [comment]
Add withPivot in the relation
https://laravel.com/docs/11.x/eloquent-relationships#retrieving-intermediate-table-columns

**Reply by Youssef-Elouasbi** [reply]
thanks for your answer 
but the problem is when we create a relation in filament like this 
php artisan make:filament-relation-manager ConventionResource enseignes nom

i specified the method inside my model (Convention) this :

public function enseignes(): BelongsToMany
{
return $this->belongsToMany(Enseigne::class, 'convention_enseignes')->withPivot('cree_par', 'modifie_par', 'proprietaire', 'date_debut', 'date_fin')->withTimestamps();
}


then when the relation was created i get the data from the enseignes table based on this intermediate table (convention_enseignes)

now what i want is a relation that get data from this table convention_enseignes i mean getting the columns and the values that will be shown in the filament table from the intermediate table ?

**Reply by leandrocfe** [reply]
Sorry, I didn't understand. Could you share a screenshot to describe what are you trying to achieve?

**Reply by Youssef-Elouasbi** [reply]
![Screenshot 2024-04-25 at 02-00-46 Edit c1 - Fidel'Up - La Fidélité à votre Style](https://github.com/filamentphp/filament/assets/79534618/5cb48dcd-ea93-455d-aad5-87e81341ca2f)


ok like you see here in this image i have a relation enseigne inside my convention edit page

now this data is coming from table enseignes but i want to get data from the intermediate table (convention_enseignes)

i have already share code for all the three modules (convention, enseigne , convention_enseignes)

this table (convention_enseignes) contain (convention_id, enseigne_id, date_debut, date_fin ....) i would like to see this data and also the form for the new button i want to be able to add value inside this table not to the enseignes table
